link.go

```diff
@@ -512,7 +509,9 @@ func (l *link) muxReceive(fr performTransfer) error {
 	debug(1, "deliveryID %d before push to receiver - deliveryCount : %d - linkCredit: %d, len(messages): %d, len(inflight): %d", l.msg.deliveryID, l.deliveryCount, l.linkCredit, len(l.messages), len(l.receiver.inFlight.m))
 	// send to receiver, this should never block due to buffering
 	// and flow control.
-	l.addUnsettled(&l.msg)
+	if l.receiverSettleMode.value() == ModeSecond {
+		l.addUnsettled(&l.msg)
+	}
```

Comment:
let's separate the optimization from the fix.
we should still add the message to the unsettled map, but we remove the condition above.
In modeFirst, the HandleMessage func is spontaneously settling the message for you, so it removes it from the map.

we can consider an optimization in a separate PR

Reply:
nevermind the comment above.
didn't realize that there was also a check added in the receiver.
I think that should be good.

Would be great to implement a test to make sure we don't have regressions in the future